types/foundry-vtt-api/types/apps/app.d.ts

```diff
@@ -0,0 +1,334 @@
+    tabs?: TabV2Options[];
+    /** A default window title string (popOut only) */
+    title?: string;
+    /** The default HTML template path to render for this Application */
+    template?: string;
+    /**
+     * A list of unique CSS selectors which target containers that should
+     * have their vertical scroll positions preserved during a re-render.
+     */
+    scrollY?: string[];
+    [key: string]: any;
+}
+
+interface RenderOptions extends Object {
+    /** The left positioning attribute */
+    left?: number;
+    /** The top positioning attribute */
+    top?: number;
+    /** The rendered width */
+    width?: number;
+    /** The rendered height */
+    height?: number;
+    /** The rendered transformation scale */
+    scale?: number;
+    /** Whether to display a log message that the Application was rendered */
+    log?: boolean;
+    /** A context-providing string which suggests what event triggered the render */
+    renderContext?: string;
+    /** The data change which motivated the render request */
+    renderData?: any;
+}
+
+interface ApplicationPosition extends Object {
+    width?: number;
+    height?: number;
+    left?: number;
+    top?: number;
+    scale?: number;
+}
+
+declare const MIN_WINDOW_WIDTH: number, MIN_WINDOW_HEIGHT: number;
+
+declare let _appId: number;
+declare let _maxZ: number;
+
+/**
+ * The standard application window that is rendered for a large variety of UI elements in Foundry VTT
+ */
+declare class Application {
+    /**
+     * The options provided to this application upon initialization
+     */
+    options: ApplicationOptions;
+
+    /**
+     * The application ID is a unique incrementing integer which is used to identify every application window
+     * drawn by the VTT
+     */
+    appId: number;
+
+    /**
+     * Track the current position and dimensions of the Application UI
+     */
+    position: ApplicationPosition;
+
+    /**
+     * An internal reference to the HTML element this application renders
+     */
+    protected _element: JQuery;
+
+    /**
+     * DragDrop workflow handlers which are active for this Application
+     */
+    protected _dragDrop: DragDrop[];
+
+    /**
+     * Tab navigation handlers which are active for this Application
+     */
+    protected _tabs: TabsV2[];
+
+    /**
+     * Track whether the Application is currently minimized
+     */
+    protected _minimized: boolean;
+
+    /**
+     * Track the render state of the Application
+     * @see {Application.RENDER_STATES}
+     */
+    protected _state: any;
+
+    /**
+     * Track the most recent scroll positions for any vertically scrolling containers
+     */
+    protected _scrollPositions: any | null;
+
+    protected static RENDER_STATES: any;
+
+    constructor(options?: ApplicationOptions);
+
+    /**
+     * Create drag-and-drop workflow handlers for this Application
+     * @return	An array of DragDrop handlers
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/foundry-vtt-api/types/apps/app.d.ts;
  Line: 125;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.